### ISSUE TITLE
don't have permission to save *.kdbx via SeafilePro

### ISSUE THREAD

**Issue by myllaeri** [keepassium/KeePassium#144]
Hi,

I store my kdbx file in a [Seafile-Server](https://github.com/haiwen/seafile-docker) and I use the [SeafilePro](https://apps.apple.com/de/app/seafile-pro/id639202512) App to access the kdbx from KeePassium on my iPhone.
opening it and using the kdbx works just fine, but when I change something and KeePassium tries to sync the kdbx it tell's me: "You don't have permission to save the file "KeyData.kdbx" in the folder "https%3a%2a%2a###SEAFILEDOMAIN###_myllaeri@###MYDOMAIN###__af69c3e4-d976-40e2-afc7-d224af12d489_%2F""

---- Diagnostic Info -----
2.319 (I) DatabaseManager.swift:224 _saveDatabase(_:dbRef:) Saving database
2.331 (I) FileKeeper.swift:868 makeBackup(nameTemplate:mode:contents:) Backup copy created OK
2.331 (I) DatabaseSaver.swift:115 save() Encrypting database
2.331 (I) Database2.swift:1016 save() Saving KP2 database
2.331 (D) Database2.swift:1023 save() Format version: v3
2.331 (D) Database2.swift:1026 save() Seeds randomized OK
2.331 (D) Database2.swift:647 deriveMasterKey(compositeKey:cipher:canUseFinalKey:) Start key derivation
3.040 (D) Database2.swift:1032 save() Key derivation OK
3.040 (D) Database2.swift:1302 toXml() Will generate XML
3.181 (D) Database2.swift:1062 save() XML generation OK
3.181 (D) Database2.swift:1206 encryptBlocksV3(to:xmlData:) Encrypting KP2v3 blocks
3.233 (D) Database2.swift:1070 save() Content encryption OK
3.233 (D) Database.swift:250 resolveReferences(allEntries:parentProgress:pendingProgressUnits:) Resolving references
3.233 (D) Database.swift:276 resolveReferences(allEntries:parentProgress:pendingProgressUnits:) References resolved OK
3.234 (I) DatabaseSaver.swift:124 save() Writing database document
3.240 (I) DatabaseManager.swift:215 startSavingDatabase() Async database saving finished
3.252 (E) FileAccessError.swift:118 make(from:fileProvider:) Failed to access the file [fileProvider: com.seafile.seafilePro.SeafFileProvider, systemError: Error Domain=NSCocoaErrorDomain Code=513 "You don’t have permission to save the file “KeyData.kdbx” in the folder “https%3A%2F%2F###SEAFILEDOMAIN###_myllaeri@###MYDOMAIN###_af69c3e4-d976-40e2-afc7-d224af12d489_%2F”." UserInfo={NSFileOriginalItemLocationKey=file:///private/var/mobile/Containers/Shared/AppGroup/4E17096D-797F-4D3B-A5FC-452A9B2F4F03/File%20Provider%20Storage/https%253A%252F%252F###SEAFILEDOMAIN###_myllaeri@###MYDOMAIN###_af69c3e4-d976-40e2-afc7-d224af12d489_%252F/KeyData.kdbx, NSURL=file:///private/var/mobile/Containers/Shared/AppGroup/4E17096D-797F-4D3B-A5FC-452A9B2F4F03/File%20Provider%20Storage/https%253A%252F%252F###SEAFILEDOMAIN###_myllaeri@###MYDOMAIN###_af69c3e4-d976-40e2-afc7-d224af12d489_%252F/KeyData.kdbx, NSFileNewItemLocationKey=file:///private/var/mobile/Containers/Data/Application/0F84A135-81BE-4104-9FDC-3CD6CF437D76/tmp/NSIRD_KeePassium_zgvai4/KeyData.kdbx, NSUnderlyingError=0x2808e05a0 {Error Domain=NSCocoaErrorDomain Code=513 "You don’t have permission to save the file “KeyData.kdbx” in the folder “NSIRD_KeePassium_zgvai4”." UserInfo={NSURL=file:///private/var/mobile/Containers/Data/Application/0F84A135-81BE-4104-9FDC-3CD6CF437D76/tmp/NSIRD_KeePassium_zgvai4/KeyData.kdbx, NSFilePath=/private/var/mobile/Containers/Data/Application/0F84A135-81BE-4104-9FDC-3CD6CF437D76/tmp/NSIRD_KeePassium_zgvai4/KeyData.kdbx, NSUnderlyingError=0x2808e0540 {Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted"}}}}]
3.254 (E) DatabaseSaver.swift:138 save() Database saving error. [message: You don’t have permission to save the file “KeyData.kdbx” in the folder “https%3A%2F%2F###SEAFILEDOMAIN###_myllaeri@###MYDOMAIN###_af69c3e4-d976-40e2-afc7-d224af12d489_%2F”.]

KeePassium v1.21.81 (iPhone, iOS 14.4)

Is there a workaround?
Does this have to do with issue #126?

**Comment by keepassium** [comment]
Hi,

I have encountered this problem a few days ago, testing with the free tier of Nihao Cloud (they are [listed](https://www.seafile.com/en/partner/) as Seafile partners).

The problem is always reproducible, but it affects other apps as well:
- First, I tried to edit a .doc file — but Word said the file was read-only. 
- Then I opened a .pdf in Files's native preview, and added some markup. After saving, the file got stuck showing the "Waiting..." status in Files.

So it seems to be something on Seafile's side.  They have two similar reports:
- https://github.com/haiwen/seafile-iOS/issues/262 was attributed to server misconfiguration
- https://github.com/haiwen/seafile-iOS/issues/251 remains a mystery since 2019... 

The workaround is to try to access your server [via WebDAV](https://keepassium.com/articles/sync-ios-keepass-with-webdav/).
